Libraries/Azure.psm1

```diff
@@ -2397,23 +2397,73 @@ function Add-AzureAccountFromSecretsFile {
 
     if ( Test-Path $secretsFile ) {
         Write-LogInfo "$secretsFile found."
-        Write-LogInfo "------------------------------------------------------------------"
-        Write-LogInfo "Authenticating Azure PS session.."
         $XmlSecrets = [xml](Get-Content $secretsFile)
         $ClientID = $XmlSecrets.secrets.SubscriptionServicePrincipalClientID
         $TenantID = $XmlSecrets.secrets.SubscriptionServicePrincipalTenantID
         $Key = $XmlSecrets.secrets.SubscriptionServicePrincipalKey
-        $pass = ConvertTo-SecureString $key -AsPlainText -Force
-        $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+        $AzureContextFilePath = $XmlSecrets.secrets.AzureContextFilePath
         $subIDSplitted = ($XmlSecrets.secrets.SubscriptionID).Split("-")
         $subIDMasked = "$($subIDSplitted[0])-xxxx-xxxx-xxxx-$($subIDSplitted[4])"
-
-        $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
-        $selectedSubscription = Select-AzureRmSubscription -SubscriptionId $XmlSecrets.secrets.SubscriptionID
+        Write-LogInfo "------------------------------------------------------------------"
+        if ($ClientID -and $Key) {
+            Write-LogInfo "Authenticating Azure PS session using Service Principal..."
+            $pass = ConvertTo-SecureString $key -AsPlainText -Force
+            $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+            $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
+        } elseif ($AzureContextFilePath) {
+            Write-LogInfo "Authenticating Azure PS session using saved context file $AzureContextFilePath..."
+            $null = Import-AzureRmContext -Path $AzureContextFilePath
+        } else {
+            $ContextFiles = Get-ChildItem -Path $PWD -Recurse | `
+                Where-Object { $_.Name.EndsWith(".json") } | Select-String -Pattern "AzureCloud" | Select Path
+            $ContextFiles = $ContextFiles.Path | Get-Unique
+            if ($ContextFiles.Count -eq 0) {
+                Write-LogWarn "No Azure authentication methods were available in Secret File."
+                Write-LogWarn "No Azure authentication context files detected in $PWD."
+
+                Write-LogWarn "Checking if current Azure powershell session is already authenticated..."
```

Comment:
needs to revise this work flow. if the user got authenticated with another sub before, but reach to this point. this is not what this script intends to do.

Reply:
@juhlee-microsoft , The priority of the authentication goes as follows - 
1. Service Principle given in the Secret File.
2. Azure Context File Path given in the secret file.
3. Azure context file found in the current directory.
4. Current authenticated session. 

Now, If user got authenticated with another subscription, and if user reaches to this point, then still we use "Select-AzureRmSubscription" command. So, if he doesn't have "Contributor" access to "Subscription Under Test", he will not be able to run LISAv2, even if the PS session is authenticated.

An authenticated user doesn't mean that he is authorized to use the subscription.

Reply:
"authorized" means here "contributor" who can create RG/VM then.